src/leaflike/bookmarks/core.clj

```diff
@@ -1,32 +1,39 @@
 (ns leaflike.bookmarks.core
-  (:require [leaflike.bookmarks.validator :refer [valid-bookmark?
-                                                  id?]]
+  (:require [leaflike.bookmarks.validator :refer [valid-bookmark?]]
             [leaflike.bookmarks.db :as bm-db]
             [leaflike.user.db :as user-db]
             [leaflike.user.auth :refer [throw-unauthorized]]
             [leaflike.utils :as utils]
-            [clojure.walk :as walk]
+            [clojure.spec.alpha :as s]
             [clojure.string :as string]))
 
 (defn- get-user
   [request]
   (let [username (:username request)]
     (first (user-db/get-member-auth-data username :id))))
 
+(defn- format-tags
+  "Convert :tags in a bookmark from a comma-separated string to a vector
+  of strings."
+  [bookmark]
+  (update bookmark :tags
+          #(if (string/blank? %)
+             nil
+             (string/split % #","))))
+
 (defn create
   [{:keys [params] :as request}]
-  (let [bookmark (-> (select-keys (walk/keywordize-keys params)
+  (let [bookmark (-> (select-keys params
                                   [:title :url :tags])
-                     (update-in [:tags] #(if (empty? %)
-                                           nil
-                                           (string/split % #","))))
+                     format-tags)
         user    (get-user request)]
```

Comment:
If this binding is moved above bookmark, you can construct the bookmark without having to shadow it later